src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -603,30 +604,37 @@ export default class MongooseAdapter {
     return words.join(" ");
   }
 
-  static getIdQueryType(idOrIds) {
-    const mode = typeof idOrIds === "string" ? "findOne" : "find";
-    let idQuery;
+  static getIdQueryType(idOrIds, Model) {
+    if (idOrIds == null) {
+      return Q([ "find", undefined ]);
+    }
+
+    const [ mode, ids ] = Array.isArray(idOrIds) ? [ "find", idOrIds ] : [ "findOne", [ idOrIds ] ];
+    const idValidationPromises = ids.map(id => this.validateId(id, Model));
 
-    if (typeof idOrIds === "string") {
-      if (!this.idIsValid(idOrIds)) {
+    return Q.all(idValidationPromises).then(() => {
+      const idQuery = { _id: mode === "find" ? { $in: ids } : ids[0] };
+      return [ mode, ids.length ? idQuery : undefined ];
```

Comment:
Wait, wouldn't the proper test be `idOrIds != null ? idQuery : undefined`? If you pass `[]` you get an empty query, if you pass `null` you get `undefined`.